pkg/gui/controllers/files_controller.go

```diff
@@ -355,6 +367,57 @@ func (self *FilesController) ignore(node *filetree.FileNode) error {
 	return self.c.Refresh(types.RefreshOptions{Scope: []types.RefreshableView{types.FILES}})
 }
 
+func (self *FilesController) exclude(node *filetree.FileNode) error {
```

Comment:
there's still some overlap between the `ignore` function and this function, so I would DRY this up some more. Lemme know if you need guidance on that

Reply:
@jesseduffield given all of the steps in both functions have to happen in the same order for both I have pulled all of it out into its own function that can be called from both `ignore` and `exclude` that way the only thing that as really happing in `ignore`,  or `exclude` is the file path check. Both the  `exclude` and `ignore` integration tests are passing. I have also added an integration test to make sure you can't exclude the `.gitignore` file :)